sample.conf

```diff
@@ -118,40 +142,45 @@ PageType	cgi
 
 #DNSChildren	0
 
-# HTMLPre defines HTML code to insert at the very beginning of the
-# file. Use it for server-side script code, like PHP.
+# HTMLPre allows code to be inserted at the very beginning of the HTML files.
+# Be careful not to include any HTML here, as it is inserted before the
+# <html> tag in the file. Use it for server-side scripting capabilities,
+# such as PHP, to insert scripting files and other directives.
```

Comment:
To me it's more clear with the sub-clause. Would you mind having it stay?

Reply:
it's fine to add detail, but "_insert scripting files_" sounds ambiguous to me in that I don't think there's such construct. Do you mean something like server-side includes? If that's what you are after, it's fine, but let's phrase it the way PHP manual describes it, so it's familiar to people.

Reply:
Well, that was taken from an older version of the very same file (not my phrasing). But yeah, if we do a Freud/Jung analysis, it gets weirder the longer one looks at it :zany_face:

> […] Use it for server-side scripting capabilities, such as PHP, to insert calls to script files and other directives.

sounds more convincing IMHO. If you agree, I change it to that.

I must admit I've never used `HTMLPre` as 1) I wasn't sure how *exactly* it is meant, and 2) had no idea what use to put it to. So just to confirm I got it right, can you give a valid example?

```shell
HTMLPre /path/to/foobar.php        # with the script having a she-bang
HTMLPre <?php echo "foobar"; ?>    # inline PHP code; can it be multi-line?
HTMLPre /usr/local/bin/prepend.sh  # a shell script, again with she-bang for the interpreter
HTMLPre fortune                    # an ELF binary spitting out a saying/quote
```

All of them correct – and what they spit out to STDOUT would be included?

Reply:
This is the structure for all those HTML configuration parameters. Their positioning didn't change from the original Webalizer, but the contents did. The original configuration allowed people to replace `DOCTYPE` and `<body>` with their own tags, which made it impossible to introduce any stable HTML content into reports, so their meaning changed.
```html
HTMLPre
<!DOCTYPE HTML>
<html>
  <head>
    ...
    HTMLHead
  </head>

  <body>
    HTMLBody
    ... page header div ...
    HTMLPost

    ... reports ...

    HTMLTail
    ... page footer div ...
    HTMLEnd
  </body>
</html>
```
Those constructs inside `<body>` allow people to wrap header/footer in their own block tags and style them differently. `HTMLPre` allows people to put any PHP, ASP, JSP, etc content, which typically would be just an include to point to a reusable PHP file, like this:
```php
HTMLPre  <?php include 'xyz.php'; ?>
```
Whatever is included, however, must be processed by the server-side scripting engine and cannot generate any output because having any characters before `DOCTYPE` invalidates HTML. In other words, having PHP `echo` in `HTMLPre` would produce invalid HTML.

If the new wording works for you, sure let's use that.

Reply:
Got it, thanks! Especially with that "struct" it got very clear (maybe copy-paste that to the documentation/Readme unless it's already there and I just missed it?). Just to make sure, I'd add parts of what you just wrote:

```html
# HTMLPre allows code to be inserted at the very beginning of the HTML files.
# Be careful not to include any HTML here, as it is inserted before the
# <!DOCTYPE> tag in the file. Use it for server-side scripting capabilities,
# such as running specific PHP code whenever the report is viewed.
# Whatever is included, however, must be processed by the server-side scripting
# engine and cannot generate any output because having any characters before <!DOCTYPE>
# invalidates HTML. In other words, having PHP echo in HTMLPre would produce invalid HTML.

# HTMLPre  <?php include 'xyz.php'; ?>
```